src/horizon_client.rs

```diff
@@ -1911,41 +1962,135 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_get_effects_for_ledger() {
-        // found by trial and error in the Stellar laboratory
-        let ledger_sequence = 125;
+    async fn get_effects_for_transactions() {
+        const ID: &str = "0000000459561504769-0000000001";
+        const PAGING_TOKEN: &str = "459561504769-1";
+        const ACCOUNT: &str = "GAIH3ULLFQ4DGSECF2AR555KZ4KNDGEKN4AFI4SU2M7B43MGK3QJZNSR";
+        const RECORD_TYPE: &str = "account_created";
+        const TYPE_I: u32 = 0;
+        const CREATED_AT: &str = "2024-02-06T17:42:48Z";
+        const STARTING_BALANCE: &str = "10000000000.0000000";
 
         let horizon_client =
             HorizonClient::new("https://horizon-testnet.stellar.org".to_string()).unwrap();
 
-        let effects_for_ledger_request =
-            EffectsForLedgerRequest::new().set_sequence(ledger_sequence);
-        let _effects_for_ledger_response = horizon_client
-            .get_effects_for_ledger(&effects_for_ledger_request)
+        let effects_for_liquidity_pools_request =
+            EffectForTransactionRequest::new().set_limit(2).unwrap();
+
+        let effects_for_liquidity_pools_response = horizon_client
+            .get_effects_for_transaction(&effects_for_liquidity_pools_request)
             .await;
 
-        println!("{:#?}", _effects_for_ledger_response);
+        assert!(effects_for_liquidity_pools_response.clone().is_ok());
+
+        assert_eq!(
+            effects_for_liquidity_pools_response
+                .clone()
+                .unwrap()
+                ._embedded()
+                .records()[0]
```

Comment:
Het ophalen van het eerste record zou ik 1 keer doen. En dan hergebruiken over alle onderliggende asserts.

Reply:
Ik zie m toch liever nu al mee in deze pr, opdat we het niet vergeten.